qiskit_nature/algorithms/excited_states_solvers/qeom.py

```diff
@@ -409,9 +414,67 @@ def _compute_excitation_energies(m_mat: np.ndarray, v_mat: np.ndarray, q_mat: np
         return excitation_energies_gap, res[1]
 
 
-class QEOMResult(AlgorithmResult):
+class QEOMResult(AlgorithmResult, collections.UserDict):
```

Comment:
I agree. We should be inline with the Qiskit result classes. I added the UserDict temporarily because there were so many other changes that I didn't want to add more stuff to this PT. I waqs going to change it in a follow-up PR but can go ahead and fix it here.